runtime/compiler/control/CompilationThread.cpp

```diff
@@ -6519,7 +6508,7 @@ TR::CompilationInfoPerThreadBase::preCompilationTasks(J9VMThread * vmThread,
          !TR::CompilationInfo::isCompiled(method) &&
          !entry->isDLTCompile() &&
          !entry->_doNotUseAotCodeFromSharedCache &&
-         reloRuntime->isRomClassForMethodInSharedCache(method) &&
+         TR_J9VMBase::get(_jitConfig, vmThread)->sharedCache()->isPointerInSharedCache(J9_CLASS_FROM_METHOD(method)->romClass) &&
```

Comment:
isn't `reloRuntime->isRomClassForMethodInSharedCache(method)` a much more compact query than `TR_J9VMBase::get(_jitConfig, vmThread)->sharedCache()->isPointerInSharedCache(J9_CLASS_FROM_METHOD(method)->romClass)`? Is there a reason or change in the future that required removing the `isRomClassForMethodInSharedCache` query?

Reply:
Ah does it have to do with the duplicated code that's there in both the SharedCache class and the RelocationRuntime class?

Reply:
Yes, we have two queries that are essentially identical but implemented in two separate classes. It hurts maintainability and makes the code less clear. One query is more than enough. `TR_J9SharedCache` seems like a good place to have it because it's a bit out of place in `RelocationRuntime`.

I agree this line is a bit long winded, but it's not the norm; in most other cases it's shorter. I can add an overload for `J9Method` that would shorten this line and would probably be more convenient. Thoughts?

Reply:
Actually, we have 3 queries that do the same thing; `CompilationInfo` has one too.

Reply:
> I agree this line is a bit long winded, but it's not the norm; in most other cases it's shorter. I can add an overload for J9Method that would shorten this line and would probably be more convenient. Thoughts?

I don't know if `J9_CLASS_FROM_METHOD(method)->romClass` --> `method` is worth the effort to make an overload. What do you think about a static method in `TR_J9SharedCache`, ie:

```
class TR_J9SharedCache : public TR_SharedCache
   {
   ...
   static bool isMethodInSharedCache(J9Method *method)
      {
      return TR_J9VMBase::get(_jitConfig, vmThread)->sharedCache()->isPointerInSharedCache(J9_CLASS_FROM_METHOD(method)->romClass)
      }
   ...
   }
```
Then the calls would just become
```
TR_J9SharedCache::isMethodInSharedCache(method)
```
And if there's a need for the query "is this class in the SCC", there could be another static method added for the class. 
```
TR_J9SharedCache:isClassInSharedCache(clazz)
```

I'm not wedded to this idea, just throwing it out there.

Reply:
I considered this, but it's hiding the `TR_J9VMBase::get(_jitConfig, vmThread)`, which is an expensive way to get you hands on an FE and should really only be used if you have no other choice IMO.

Usually we have an FE handy, or a `Compilation` or `CodeGenerator` which we can get an FE from cheaply so people should prefer that. In this series of patches I've only used `::get()` in places where it was necessary. This method in particular is a mess with its FE usage and could use a serious cleanup, but I didn't go that far and just followed suit with the rest of the code to not complicate things further.

Reply:
That's fair, I'm fine leaving it as is.